pkg/controllers/systemd-etcd.go

```diff
@@ -0,0 +1,154 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/openshift/microshift/pkg/config"
+	"github.com/openshift/microshift/pkg/util/cryptomaterial"
+	"k8s.io/klog/v2"
+)
+
+type ExtEtcdService struct {
+	NodeName            string
+	DataDir             string
+	CertFile            string
+	KeyFile             string
+	TrustedCAFile       string
+	InitialUrl          string
+	PeerCertFile        string
+	PeerKeyFile         string
+	PeerTrustedCAFile   string
+	AdvertiseClientUrls string
+	ListenClientUrls    string
+	ListenPeerUrls      string
+	ListenMetricsUrls   string
+}
+
+func NewExtEtcd(cfg *config.MicroshiftConfig) *ExtEtcdService {
+	s := &ExtEtcdService{}
+	s.configure(cfg)
+	return s
+}
+
+func (s *ExtEtcdService) Name() string           { return "etcd" }
+func (s *ExtEtcdService) Dependencies() []string { return []string{} }
+
+func (s *ExtEtcdService) configure(cfg *config.MicroshiftConfig) {
+	s.NodeName = cfg.NodeName
+
+	certsDir := cryptomaterial.CertsDirectory(cfg.DataDir)
+	s.DataDir = filepath.Join(s.DataDir, s.Name())
+
+	caCertFile := cryptomaterial.UltimateTrustBundlePath(certsDir)
+	etcdCertDir := filepath.Join(certsDir, s.Name())
+
+	s.CertFile = filepath.Join(etcdCertDir, "etcd-serving.crt")
+	s.KeyFile = filepath.Join(etcdCertDir, "etcd-serving.key")
+	s.TrustedCAFile = caCertFile
+
+	s.PeerCertFile = filepath.Join(etcdCertDir, "etcd-peer.crt")
+	s.PeerKeyFile = filepath.Join(etcdCertDir, "etcd-peer.key")
+	s.PeerTrustedCAFile = caCertFile
```

Comment:
You'll want to fix these with the functions from https://github.com/openshift/microshift/blob/main/pkg/util/cryptomaterial/certpaths.go#L118-L126

See https://github.com/openshift/microshift/blob/main/pkg/controllers/etcd.go#L82-L88